app/src/main/java/org/literacyapp/authentication/AuthenticationHelper.java

```diff
@@ -0,0 +1,67 @@
+package org.literacyapp.authentication;
+
+import android.content.Context;
+import android.util.Log;
+
+import org.apache.commons.math3.analysis.function.StepFunction;
+import org.literacyapp.LiteracyApplication;
+import org.literacyapp.dao.AuthenticationEventDao;
+import org.literacyapp.dao.DaoSession;
+import org.literacyapp.model.Device;
+import org.literacyapp.model.Student;
+import org.literacyapp.model.analytics.AuthenticationEvent;
+import org.literacyapp.receiver.BootReceiver;
+import org.literacyapp.util.DeviceInfoHelper;
+import org.literacyapp.util.StudentUpdateHelper;
+
+import java.util.Calendar;
+import java.util.List;
+
+/**
+ * Created by sladomic on 30.12.16.
+ */
+
+public class AuthenticationHelper {
+    private static final double[] countSteps = { 0, 5, 10, 15, 20 };
+    private static final double[] exponentialDelayStepsInMinutes = { 30, 60, 120, 240, 360 };
```

Comment:
Another question is this: how long time should it take to reach the maximum delay? Now it might take 30 + 60 + 120 + 240 + 360 = 810 minutes for one Student to reach the 360 minute delay, right? All in just one day. So maybe it's too aggressive to increase the delay so quickly, especially since it's likely to see different Students using a tablet over a period of 2 days. Tablet switching is more probable to happen each morning than during the day.

How about this: What if we follow the same formula as above, but we only increase the delay if the previous delay adjustment was performed the day before or earlier (not during the same day)? This would require storing a timestamp for the last time the delay was adjusted.